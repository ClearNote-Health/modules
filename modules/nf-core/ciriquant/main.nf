process CIRIQUANT {
    tag "$meta.id"
    label 'process_high'

    conda "bioconda::ciriquant=1.1.2"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/ciriquant:1.1.2--pyhdfd78af_2' :
        'quay.io/biocontainers/ciriquant:1.1.2--pyhdfd78af_2' }"

    input:
    tuple val(meta), path(reads)
    path gtf
    path fasta
    path bwa
    path hisat2

    output:
    tuple val(meta), path("${prefix}/${prefix}.gtf"), emit: gtf
    path("${prefix}")                               , optional: true, emit: intermediates_directory
    path "versions.yml"                             , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    bwa_prefix = fasta.toString() == 'genome.fa' ? fasta.toString() : fasta.toString() - ~/.(fa|fasta)$/
    hisat2_prefix = fasta.toString() - ~/.(fa|fasta)$/
    fasta_path = fasta.toRealPath()
    gtf_path = gtf.toRealPath()
    bwa_path = bwa.toRealPath()
    hisat2_path = hisat2.toRealPath()
    """
    BWA=`which bwa`
    HISAT2=`which hisat2`
    STRINGTIE=`which stringtie`
    SAMTOOLS=`which samtools`

    touch travis.yml
    printf "name: ciriquant\ntools:\n  bwa: \$BWA\n  hisat2: \$HISAT2\n  stringtie: \$STRINGTIE\n  samtools: \$SAMTOOLS\n\nreference:\n  fasta: ${fasta_path}\n  gtf: ${gtf_path}\n  bwa_index: ${bwa_path}/${bwa_prefix}\n  hisat_index: ${hisat2_path}/${hisat2_prefix}" >> travis.yml

    CIRIquant \\
        -t ${task.cpus} \\
        -1 ${reads[0]} \\
        -2 ${reads[1]} \\
        --config travis.yml \\
        -o ${prefix} \\
        -p ${prefix} \\
        $args

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        ciriquant: \$(echo \$(CIRIquant --version 2>&1) | sed 's/CIRIquant //g' )
    END_VERSIONS
    """
}
