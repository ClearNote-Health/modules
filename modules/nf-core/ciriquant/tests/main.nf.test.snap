{
    "chrI": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "fust1",
                            "single_end": false
                        },
                        "fust1.gtf:md5,bae6b703dd914cb29c9a6a5537590b13"
                    ]
                ],
                "1": [
                    [
                        "CIRIerror.log:md5,d41d8cd98f00b204e9800998ecf8427e",
                        [
                            "fust1.sorted.bam:md5,597b93572c2f28baf2c4ca1613c7de01",
                            "fust1.sorted.bam.bai:md5,b446d5fe2197aa56ccf62f4e02b7daac"
                        ],
                        [
                            "fust1.ciri:md5,ad8d1f86c313898c8a1db0f44716eb15",
                            "fust1_denovo.sorted.bam:md5,f53576a9f828e2a11dbc3e9c92f79e56",
                            "fust1_denovo.sorted.bam.bai:md5,4dd41bfc216ae7f3d6f39d8dcf549ddc",
                            "fust1_index.1.ht2:md5,615b3eb3e7f1bf2d0db1d76ad074ad47",
                            "fust1_index.2.ht2:md5,de058f777657c8980399a2ba75e87f4f",
                            "fust1_index.3.ht2:md5,3608c8d83f4f71db4ea5f82c9126eb82",
                            "fust1_index.4.ht2:md5,8274a66a4bcce6431b2cc9f76218d503",
                            "fust1_index.5.ht2:md5,ff3d3ecce372af6227ba34c5864814df",
                            "fust1_index.6.ht2:md5,7d1fd2da909974b2e5fc8365caac7cd2",
                            "fust1_index.7.ht2:md5,9013eccd91ad614d7893c739275a394f",
                            "fust1_index.8.ht2:md5,33cdeccccebe80329f1fdbee7f5874cb",
                            "fust1_index.fa:md5,4ed76e85d398354e21200ff373822771"
                        ],
                        "fust1.bed:md5,36cb4ebdb1de162f6c6a7e6dd7b1de97",
                        "fust1.gtf:md5,bae6b703dd914cb29c9a6a5537590b13",
                        "fust1.log:md5,f9fada91fd19538ce8ed515963a1d76f"
                    ]
                ],
                "2": [
                    "versions.yml:md5,3cd3fb4b1cf0b04df6bcef590678d8d7"
                ],
                "gtf": [
                    [
                        {
                            "id": "fust1",
                            "single_end": false
                        },
                        "fust1.gtf:md5,bae6b703dd914cb29c9a6a5537590b13"
                    ]
                ],
                "results": [
                    [
                        "CIRIerror.log:md5,d41d8cd98f00b204e9800998ecf8427e",
                        [
                            "fust1.sorted.bam:md5,597b93572c2f28baf2c4ca1613c7de01",
                            "fust1.sorted.bam.bai:md5,b446d5fe2197aa56ccf62f4e02b7daac"
                        ],
                        [
                            "fust1.ciri:md5,ad8d1f86c313898c8a1db0f44716eb15",
                            "fust1_denovo.sorted.bam:md5,f53576a9f828e2a11dbc3e9c92f79e56",
                            "fust1_denovo.sorted.bam.bai:md5,4dd41bfc216ae7f3d6f39d8dcf549ddc",
                            "fust1_index.1.ht2:md5,615b3eb3e7f1bf2d0db1d76ad074ad47",
                            "fust1_index.2.ht2:md5,de058f777657c8980399a2ba75e87f4f",
                            "fust1_index.3.ht2:md5,3608c8d83f4f71db4ea5f82c9126eb82",
                            "fust1_index.4.ht2:md5,8274a66a4bcce6431b2cc9f76218d503",
                            "fust1_index.5.ht2:md5,ff3d3ecce372af6227ba34c5864814df",
                            "fust1_index.6.ht2:md5,7d1fd2da909974b2e5fc8365caac7cd2",
                            "fust1_index.7.ht2:md5,9013eccd91ad614d7893c739275a394f",
                            "fust1_index.8.ht2:md5,33cdeccccebe80329f1fdbee7f5874cb",
                            "fust1_index.fa:md5,4ed76e85d398354e21200ff373822771"
                        ],
                        "fust1.bed:md5,36cb4ebdb1de162f6c6a7e6dd7b1de97",
                        "fust1.gtf:md5,bae6b703dd914cb29c9a6a5537590b13",
                        "fust1.log:md5,f9fada91fd19538ce8ed515963a1d76f"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,3cd3fb4b1cf0b04df6bcef590678d8d7"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-07T10:02:57.947633682"
    },
    "chrI - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "fust1",
                            "single_end": false
                        },
                        "fust1.gtf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        "fust1.gtf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,3cd3fb4b1cf0b04df6bcef590678d8d7"
                ],
                "gtf": [
                    [
                        {
                            "id": "fust1",
                            "single_end": false
                        },
                        "fust1.gtf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "results": [
                    [
                        "fust1.gtf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,3cd3fb4b1cf0b04df6bcef590678d8d7"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-07T10:03:42.746010943"
    }
}